index.js

```diff
@@ -269,10 +269,26 @@ var PgDriver = Base.extend({
         var constraint = [],
             cb;
 
+        if ((spec.unique || spec.primaryKey) && spec.autoIncrement) {
```

Comment:
What if I just called log.warn with 'SMALLINT AutoIncrement requires Postgres >= 9.2' all the time inside that case.